### ISSUE TITLE
No module named 'jupyter_server'

### ISSUE THREAD

**Issue by sigurdurb** [ipython/ipyparallel#807]
I am running Jupyter notebooks on a Jupyterhub instance.
IPyparallel insists I install jupyter_server.
I put the ipyparallel serverextension to false in /opt/conda/etc/jupyter/jupyter_server_config.d/ipyparallel.json
while it is set to true in /opt/conda/etc/jupyter/jupyter_notebook_config.d/ipyparallel.json
yet this module error keeps on happening.

Should it not tell it to use just jupyter notebook server if I only have the notebook extension set to true? What should I do? Is there some way to use ipyparallel without installing jupyter_server ?

From jupyterhub log:
```
[W 2023-05-13 17:13:06.301 SingleUserNotebookApp notebookapp:2040] Error loading server extension ipyparallel
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.8/site-packages/notebook/notebookapp.py", line 2036, in init_server_extensions
        func(self)
      File "/opt/conda/lib/python3.8/site-packages/ipyparallel/__init__.py", line 97, in _load_jupyter_server_extension
        from .nbextension.handlers import load_jupyter_server_extension
      File "/opt/conda/lib/python3.8/site-packages/ipyparallel/nbextension/handlers.py", line 7, in <module>
        from jupyter_server.utils import url_path_join as ujoin
    ModuleNotFoundError: No module named 'jupyter_server'
```

**Comment by minrk** [comment]
I would definitely recommend installing Jupyter Server. Notebook 7 will require it, so this situation will go away, but it is definitely a bug to not handle the notebook-only case while that's still supported. That should be fixed by #808.

If you really need a workaround without installing anything, I believe you can do this in your jupyter_notebook_config.py:

```python
import sys

try:
    import jupyter_server.utils
except ImportError:
    import notebook.utils
    # inject notebook.utils as jupyter_server.utils
    sys.modules["jupyter_server"] = notebook
    sys.modules["jupyter_server.utils"] = notebook.utils
```

**Comment by sigurdurb** [comment]
Yes, I ended up installing `jupyter_server` shortly after. Seems to work fine so far if there is anybody wondering.
